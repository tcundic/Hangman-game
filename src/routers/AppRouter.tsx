import React from "react";
import {BrowserRouter, Navigate, Route, Routes} from "react-router-dom";
import Header from "../components/Header";
import App from "../components/App";

const AppRouter = () => (
    <React.StrictMode>
        <Header />
        <BrowserRouter>
            <Routes>
                <Route path="/" element={<App/>}/>
                <Route path="*" element={<Navigate to="/"/>}/>
            </Routes>
        </BrowserRouter>
    </React.StrictMode>
);

export default AppRouter;
