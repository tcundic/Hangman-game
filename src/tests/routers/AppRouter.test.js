import React from "react";
import { shallow } from "enzyme";
import AppRouter from "../../routers/AppRouter";

test('should render AppRouter correctly', () => {
    const wrapper = shallow(<AppRouter />);
    expect(wrapper).toMatchSnapshot();
});
