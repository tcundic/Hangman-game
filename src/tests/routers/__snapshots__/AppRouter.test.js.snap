// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AppRouter correctly 1`] = `
<StrictMode>
  <Header />
  <BrowserRouter>
    <Routes>
      <Route
        element={<Memo(Connect(App)) />}
        path="/"
      />
      <Route
        element={<Memo(Connect(Play)) />}
        path="/play"
      />
      <Route
        element={<Memo(Connect(Highscores)) />}
        path="/highscores"
      />
      <Route
        element={
          <Navigate
            to="/"
          />
        }
        path="*"
      />
    </Routes>
  </BrowserRouter>
</StrictMode>
`;
